airflow/providers/amazon/aws/transfers/glacier_to_gcs.py

```diff
@@ -0,0 +1,103 @@
+
+    :param vault_name: the Glacier vault on which job is executed
+    :type vault_name: string
+    :param bucket_name: the Google Cloud Storage bucket where the data will be transferred
+    :type bucket_name: str
+    :param object_name: the name of the object to check in the Google cloud
+        storage bucket.
+    :type object_name: str
+    :param gzip: option to compress local file or file data for upload
+    :type gzip: bool
+    :param google_impersonation_chain: Optional Google service account to impersonate using
+        short-term credentials, or chained list of accounts required to get the access_token
+        of the last account in the list, which will be impersonated in the request.
+        If set as a string, the account must grant the originating account
+        the Service Account Token Creator IAM role.
+        If set as a sequence, the identities from the list must grant
+        Service Account Token Creator IAM role to the directly preceding identity, with first
+        account from the list granting this role to the originating account (templated).
+    :type google_impersonation_chain: Union[str, Sequence[str]]
+    """
+
+    template_fields = ("vault_name", "bucket_name", "object_name")
+
+    @apply_defaults
+    def __init__(
+        self,
+        *,
+        aws_conn_id="aws_default",
+        vault_name: str,
+        gcp_conn_id="google_cloud_default",
+        delegate_to=None,
+        google_impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        bucket_name: str,
+        object_name: str,
+        gzip: bool,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.aws_conn_id = aws_conn_id
+        self.vault_name = vault_name
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.impersonation_chain = google_impersonation_chain
+        self.bucket_name = bucket_name
+        self.object_name = object_name
+        self.gzip = gzip
+
+    def execute(self, context):
+        glacier_hook = GlacierHook(aws_conn_id=self.aws_conn_id)
+        gcs_hook = GCSHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.impersonation_chain,
+        )
+        job_id = glacier_hook.retrieve_inventory(vault_name=self.vault_name)
+        # glacier_data = glacier_hook.retrieve_inventory_results(
+        # vault_name=self.vault_name, job_id=job_id["jobId"])
+        with tempfile.NamedTemporaryFile() as temp_file:
+            glacier_data = glacier_hook.retrieve_inventory_results(
+                vault_name=self.vault_name, job_id=job_id["jobId"]
+            )
+            temp_file.write(glacier_data)
+            temp_file.flush()
+            gcs_hook.upload(
+                bucket_name=self.bucket_name,
+                object_name=self.object_name,
+                filename=temp_file.name,
+                gzip=self.gzip,
+            )
```

Comment:
It is important that the user of this operators know that it works like that.

I mean copying files to the local instance must be done with caution. Airflow may not be running on a big machine with sufficient memory, etc.

So I would suggest that you just add a `.. note::` to the docs of the operator at the top of the class. And to the howto docs as well. WDYT?

Reply:
> So I would suggest that you just add a .. note:: to the docs of the operator at the top of the class. And to the howto docs as well. WDYT?

@feluelle I don't think we have such note anywhere in transfer operators 😄

Reply:
Yesterday with Tomek we discussed about handling downloading object from Glacier and I think this part of operator would be changed a bit.

Reply:
Ok, I will go with `.. note::` and I will add note in `howto` section about this operator.

Reply:
@turbaszek s3 to redshift for example can copy it over without the need of local storaging

But I think you are right that we do not have that in most cases.